app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -94,19 +93,61 @@ protected void onCreate(final Bundle savedInstanceState) {
     }
 
     /**
-     * Called when the floating action button is called
+     * {@inheritDoc}
+     */
+    @Override
+    protected void onResume() {
+        super.onResume();
+        try {
+            this.contacts = serviceLocator.getDatabase().getAllContacts();
+        } catch (final IOException e) {
+            LOGGER.error("onResume in MainActivity threw an IOException", e);
+        }
+    }
+
+    /**
+     * Gets triggered when the Bluetooth button is clicked.
      *
-     * @param view The floating action buttion (fab)
+     * @param view - the ImageButton
      */
-    public void onClickFab(final View view) {
+    public void onBluetoothButtonClick(final View view) {
         LOGGER.info("Bluetooth button clicked");
-        startBluetoothPairing();
+        final Intent intent = new Intent(this, ActivateBluetoothActivity.class);
+        intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+        startActivity(intent);
+    }
+
+
+    /**
+     * Gets triggered when the NFC button is clicked.
+     *
+     * @param view - the ImageButton
+     */
+    public void onNFCButtonClick(final View view) {
+        LOGGER.info("NFC button clicked");
+        final Intent intent = new Intent(this, NFCActivity.class);
+        intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+        startActivity(intent);
+    }
+
+    /**
+     * Gets triggered when the QR button is clicked.
+     *
+     * @param view - the ImageButton
+     */
+    public void onQRButtonClicked(final View view) {
+        LOGGER.info("QR button clicked");
+        final Intent intent = new Intent(this, QRActivity.class);
+        intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+        startActivity(intent);
     }
 
     /**
      * Switches the sorting mode.
+     *
+     * @param view The sort floating action button that was clicked
      */
-    public void onSortButtonClick(final View view) {
+    public void onSortButtonClicked(final View view) {
```

Comment:
Could you be consistent with naming these methods, sometimes you use `onClick` and sometimes `onClicked`...

I prefer `onClick` because it is active and shorter then `onClicked`. Also I think we've used `onClick` so far in other classes as well...